RCTAppleHealthKit/RCTAppleHealthKit.m

```diff
@@ -329,4 +353,99 @@ - (void)getModuleInfo:(NSDictionary *)input callback:(RCTResponseSenderBlock)cal
     callback(@[[NSNull null], info]);
 }
 
+#pragma mark Meals
+-(void)reportHealthKitMeal:(NSDictionary *)input resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject
+{
+    NSString *id = [input objectForKey:@"id"];
+    NSString *name = [input objectForKey:@"name"];
+    NSString *dateTimeString = [input objectForKey:@"dateTime"];
+    NSString *type = [input objectForKey:@"type"];
+    int energy = [[input valueForKey:@"energy"] integerValue];
+    NSDictionary *macros = [input objectForKey:@"macros"];
+    int fat = [[macros valueForKey:@"macros"] integerValue];
+    int fiber = [[macros valueForKey:@"fiber"] integerValue];
+    int sugar = [[macros valueForKey:@"sugar"] integerValue];
+    int carbs = [[macros valueForKey:@"carbs"] integerValue];
+    int protein = [[macros valueForKey:@"protein"] integerValue];
+    
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    dateFormatter.dateFormat = @"yyyy-MM-dd'T'HH:mm:ss.SSS";
+    NSDate *mealDate = [dateFormatter dateFromString:dateTimeString];
+    
+    NSDictionary *metadata = [[NSDictionary alloc] initWithObjectsAndKeys:id, @"mealId", name, @"recipe", type, @"type", nil];
```

Comment:
Okay, yes...we have to make sure that none of `id`, `name`, `type` are non nil.

Also this can be a little cleaner with:
```
NSDictionary *metadata = @{@"mealId": id, @"recipe": name, @"type": type}
```

(i know, objective-c is gross)